### ISSUE TITLE
Improving performance by breaking out of forEach loops

### ISSUE THREAD

**Issue by abdulapopoola** [kangax/jscritic#11]
Would it be better to just do a quick check for variables, set the flag to true and return immediately?

I see most of the forEach loops keep overwriting the existing values and are redundant. For example: 
(result.errors || []).forEach(function(e) {
    if (e.code === 'W060') {
      jscriticResult.hasDocumentWrite = true;
    }
  });

A search for the first error code 'W060' would achieve the same purpose.

**Comment by kangax** [comment]
Yeah, it could use some optimization. We can't return from forEach though.

**Comment by abdulapopoola** [comment]
Exactly; throwing an exception might work but it looks ugly. The other alternative would be to use a for loop and exit fast.

I noticed it took some time when I copy-pasted jQuery into the window.
